SonarQube.TeamBuild.PreProcessor/SonarWebService.cs

```diff
@@ -32,45 +30,42 @@ public SonarWebService(IDownloader downloader, string server, string language, s
             {
                 throw new ArgumentNullException("server");
             }
-            if (string.IsNullOrWhiteSpace(language))
-            {
-                throw new ArgumentNullException("language");
-            }
-            if (string.IsNullOrWhiteSpace(repository))
-            {
-                throw new ArgumentNullException("repository");
-            }
             
             Downloader = downloader;
             Server = server.EndsWith("/", StringComparison.OrdinalIgnoreCase) ? server.Substring(0, server.Length - 1) : server;
-            Language = language;
-            Repository = repository;
         }
 
         /// <summary>
         /// Get the name of the quality profile (of the given language) to be used by the given project key
         /// </summary>
-        public string GetQualityProfile(string projectKey)
+        public bool TryGetQualityProfile(string projectKey, string language, out string qualityProfile)
         {
             string contents;
-            var ws = GetUrl("/api/profiles/list?language={0}&project={1}", Language, projectKey);
+            var ws = GetUrl("/api/profiles/list?language={0}&project={1}", language, projectKey);
             if (!Downloader.TryDownloadIfExists(ws, out contents))
             {
-                ws = GetUrl("/api/profiles/list?language={0}", Language);
+                ws = GetUrl("/api/profiles/list?language={0}", language);
                 contents = Downloader.Download(ws);
             }
             var profiles = JArray.Parse(contents);
-            // TODO What is profiles is empty?
```

Comment:
If the VB.NET plugin has never been installed, 0 profiles will be returned, so this limitation had to be fixed.